### PR TITLE
[15.0] [MIG] `website_sale_cart_expire`

### PR DESCRIPTION
**Pull request by ivantodorovich**
Standard migration plus a few extra feature and fixes, each in a separate commit to ease review and/or backporting

### CODE REVIEW

**Review by petrus-v — APPROVED** [review]
Thanks for this migration !

Code review looks good to me.

File: website_sale_cart_expire/models/website.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        expire_date = fields.Datetime.now() - timedelta(hours=self.cart_expire_delay)
+        return [
+            ("website_id", "=", self.id),
+            ("state", "=", "draft"),
+            ("write_date", "<=", expire_date),
+            # We don't want to cancel carts that are already in payment.
+            "|",
+            ("transaction_ids", "=", False),
+            "!",
+            ("transaction_ids.state", "in", ["pending", "authorized", "done"]),
```
Comment (petrus-v) [inline website_sale_cart_expire/models/website.py:35]: It's common to see the negative term `"!"` used directly as it's translated by odoo from operator, so I would suggest but won't lock merging process for that 

```suggestion
            ("transaction_ids.state", "not in", ["pending", "authorized", "done"]),
```

**Review by petrus-v — APPROVED** [review]

File: website_sale_cart_expire/models/website.py (hunk 1 above)
Comment (ivantodorovich) [reply website_sale_cart_expire/models/website.py:35]: In this case it's not the same, though.

Using the `!` operator it means: _Orders that don't have any transaction in these states_
Using the `not in` operator it means: _Orders having transactions in different states than these_

**Review by ioans73 — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**